sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/state/StatePartitionReader.scala

```diff
@@ -166,16 +184,22 @@ class StatePartitionReader(
         stateVariableInfoOpt match {
           case Some(stateVarInfo) =>
             val stateVarType = stateVarInfo.stateVariableType
-            val hasTTLEnabled = stateVarInfo.ttlEnabled
```

Comment:
Is it intentional to remove out the readability of TTL functionality for value state? Just wanted to know whether we give up functionality in certain reason, or you want to defer this till we have every types in support range and think about UX which works for every types.

Reply:
Yea - for list state, we can't have a separate column. So in that case, we have to embed it as part of the value itself. So trying to make it uniform for value state as well

Reply:
But say, if we explode elements in list state (and map state), we will be able to add TTL column for them and the schema for the value would be very similar (only user key in map state). Do I understand correctly?

Reply:
As commented in other comment thread, I'm OK with this change, but let's see how we could deal with TTL in general. We can defer the decision till we are seeing a full picture of this (among state types).

Reply:
OK, I see what you are doing now. It's just showing the value schema as passthrough, so TTL will be just a part of value if it's described. OK.

Reply:
Correct yea